paywall/src/data-iframe/walletWrapper.ts

```diff
@@ -0,0 +1,74 @@
+import { WalletService } from '@unlock-protocol/unlock-js'
+import {
+  WalletServiceType,
+  TransactionDefaults,
+} from './blockchainHandler/blockChainTypes'
+import { normalizeLockAddress } from '../utils/normalizeAddresses'
+import { POLLING_INTERVAL } from '../constants'
+
+// There are slight differences between what WalletService emits and
+// what our other code expects, this piece massages the data so that
+// they match.
+export const formatTransaction = (
+  hash: string,
+  from: string,
+  to: string,
+  input: string,
+  type: string,
+  status: string
+): TransactionDefaults => {
+  const normalizedTo = normalizeLockAddress(to)
+  const normalizedFrom = normalizeLockAddress(from)
+  return {
+    hash,
+    from: normalizedFrom,
+    for: normalizedFrom,
+    to: normalizedTo,
+    input,
+    type,
+    status,
+    blockNumber: Number.MAX_SAFE_INTEGER,
+  }
+}
+
+// hideous
+type NewTransactionArgs = [string, string, string, string, string, string]
```

Comment:
Not a deal breaker considering where we started from with this, but we should probably revisit this at some point

Reply:
Definitely agree. Until we have a good picture of the downstream consumer expectations I’m trying to avoid messing with the formatting too much (once we start using this iframe for real we’ll be able to make larger changes with confidence thanks to the integration tests)